tests/integration_tests/functional/test_error_code.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenarios for Firecracker signal handling."""
+
+import os
+import time
+import platform
+import pytest
+
+
+@pytest.mark.skipif(
+    platform.machine() == "x86_64",
+    reason="We only test with a kernel for aarch64"
+)
+def test_enosys_error_code(test_microvm_with_ssh):
+    """Test that ENOSYS error_code doesn't kill the microVM."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+
+    # We don't need to monitor the memory for this test.
+    test_microvm.memory_monitor = None
+
+    test_microvm.kernel_file = \
+        test_microvm.kernel_file[:-11] + "enosys_vmlinux.bin"
```

Comment:
as far as I can tell, this is essentially pop()-ing the last identifier from the `kernel_file` path and replacing it . the logic is brittle and we should replace it with some `os.path` method calls

Reply:
Instead of adding another kernel binary, can we just set the first instruction that is going to be executed to something that cannot be emulated and results in ENOSYS ?